Assets/_Application/Layers/LayerTypes/HierarchicalObject/HierarchicalObjectLayerGameObject.cs

```diff
@@ -123,6 +127,33 @@ private void UpdateScale(Vector3 newScale)
                 transform.localScale = newScale;
         }
 
+        public void SnapToGround()
+        {
+            Vector3 heightExtent = new Vector3(0, Bounds.Size.ToUnity().y * 0.5f, 0);
+            Vector3 pivotOffset = Bounds.Center.ToUnity() - transform.position;
+            Vector3 startPosition = WorldTransform.Coordinate.ToUnity() - heightExtent;
+            Vector3 previousPosition = WorldTransform.Coordinate.ToUnity();
+            OpticalRaycaster raycaster = FindAnyObjectByType<OpticalRaycaster>();
+            raycaster.GetWorldPointFromDirectionAsync(startPosition, Vector3.down, (w, h) =>
+            {
+                if (h)
+                    transform.position = w + heightExtent - pivotOffset;
+                else
+                {
+                    //object is below ground? check if we can snap in the up direction
+                    transform.position = previousPosition;
+                    startPosition = WorldTransform.Coordinate.ToUnity() + heightExtent;
+                    raycaster.GetWorldPointFromDirectionAsync(startPosition, Vector3.up, (ww, hh) =>
+                    {
+                        if (hh)
+                            transform.position = ww - heightExtent - pivotOffset;
+                        else //TODO the default fallback position when no hits are found is at the 0 plane, needs to be replaced with the texture height feature
```

Comment:
Kan je niet de ObjectPlacementUtility als fallback gebruiken? Die is ook niet perfect - maar dan heb je iig geen custom code en kunnen we die centraal aanpassen